src/SQLite.cs

```diff
@@ -1946,77 +1941,440 @@ protected internal void Dispose()
+	{
+		public ColumnMappingFromAttributes(PropertyInfo prop, CreateFlags createFlags = CreateFlags.None) : base(prop)
+		{
+			var colAttr = (ColumnAttribute)prop.GetCustomAttributes(typeof(ColumnAttribute), true).FirstOrDefault();
+
+			Name = colAttr == null ? prop.Name : colAttr.Name;
+			//If this type is Nullable<T> then Nullable.GetUnderlyingType returns the T, otherwise it returns null, so get the actual type instead
+			ColumnType = Nullable.GetUnderlyingType(prop.PropertyType) ?? prop.PropertyType;
+			Collation = Orm.Collation(prop);
+
+			IsPK = Orm.IsPK(prop) ||
+				(((createFlags & CreateFlags.ImplicitPK) == CreateFlags.ImplicitPK) &&
+					 string.Compare(prop.Name, Orm.ImplicitPkName, StringComparison.OrdinalIgnoreCase) == 0);
+
+			var isAuto = Orm.IsAutoInc(prop) || (IsPK && ((createFlags & CreateFlags.AutoIncPK) == CreateFlags.AutoIncPK));
+			IsAutoGuid = isAuto && ColumnType == typeof(Guid);
+			IsAutoInc = isAuto && !IsAutoGuid;
+
+			Indices = Orm.GetIndices(prop);
+			if (!Indices.Any()
+				&& !IsPK
+				&& ((createFlags & CreateFlags.ImplicitIndex) == CreateFlags.ImplicitIndex)
+				&& Name.EndsWith(Orm.ImplicitIndexSuffix, StringComparison.OrdinalIgnoreCase)
+				)
 			{
-				_prop.SetValue (obj, val, null);
+				Indices = new IndexedAttribute[] { new IndexedAttribute() };
+			}
+			IsNullable = !(IsPK || Orm.IsMarkedNotNull(prop));
+			MaxStringLength = Orm.MaxStringLength(prop);
+
+			StoreAsText = prop.PropertyType.GetTypeInfo().GetCustomAttribute(typeof(StoreAsTextAttribute), false) != null;
+		}
+	}
+
+	public static class TableMappingFluentExtensions
+	{
+		public static PropertyInfo AsPropertyInfo<TEntity>(this Expression<Func<TEntity, object>> property)
+		{
+			Expression body = property.Body;
+			var operand = (body as UnaryExpression)?.Operand as MemberExpression;
+			if (operand != null) {
+				body = operand;
 			}
+			return (body as MemberExpression)?.Member as PropertyInfo;
+		}
+
+		public static void AddPropertyValue<T, TEntity>(this Dictionary<PropertyInfo, T> dict, Expression<Func<TEntity, object>> property, T value)
+		{
+			var prop = AsPropertyInfo(property);
+			dict[prop] = value;
+		}
+
+		public static void AddProperty<TEntity>(this List<PropertyInfo> list, Expression<Func<TEntity, object>> property)
+		{
+			var prop = AsPropertyInfo(property);
+			if (!list.Contains(prop)) {
+				list.Add(prop);
+			}
+		}
 
-			public object GetValue (object obj)
+		public static void AddProperties<TEntity>(this List<PropertyInfo> list, Expression<Func<TEntity, object>>[] properties)
+		{
+			foreach (var property in properties) {
+				AddProperty(list, property);
+			}
+		}
+
+		public static T GetOrDefault<T>(this Dictionary<PropertyInfo, T> dict, PropertyInfo key, T defaultValue = default(T))
+		{
+			if (dict.ContainsKey(key)) {
+				return dict[key];
+			}
+			return defaultValue;
+		}
+	}
+
+	public class TableMappingBuilder<TEntity>
+	{
+		SQLiteConnection _sqlite;
```

Comment:
I do not want a pointer back to the connection. I'm trying to keep Mappings separate from connections.